quarkus/addons/camel/deployment/src/main/java/org/kie/kogito/addons/quarkus/camel/deployment/CamelFunctionStaticValidator.java

```
@@ -0,0 +1,100 @@
+
+import java.util.Objects;
+
+import org.kie.kogito.addons.quarkus.camel.runtime.CamelFunctionArgs;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import io.serverlessworkflow.api.Workflow;
+import io.serverlessworkflow.api.actions.Action;
+import io.serverlessworkflow.api.functions.FunctionRef;
+import io.serverlessworkflow.api.interfaces.State;
+import io.serverlessworkflow.api.states.CallbackState;
+import io.serverlessworkflow.api.states.ForEachState;
+import io.serverlessworkflow.api.states.OperationState;
+
+/**
+ * Static validation for Workflow DSL in build time.
+ */
+public final class CamelFunctionStaticValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CamelFunctionStaticValidator.class);
+
+    private CamelFunctionStaticValidator() {
+    }
+
+    /**
+     * Validation for the Function Definition within the given Workflow
+     *
+     * @param workflow the workflow reference
+     * @param functionDefName the name of the function definition
+     * @throws IllegalArgumentException if the referenced function has more than one argument in the call
+     */
+    public static void validateFunctionDefinition(final Workflow workflow, final String functionDefName) {
+        for (State state : workflow.getStates()) {
+            switch (state.getType()) {
+                case OPERATION:
+                    ((OperationState) state).getActions().forEach(action -> validateAction(action, functionDefName));
+                    break;
+                case FOREACH:
+                    ((ForEachState) state).getActions().forEach(action -> validateAction(action, functionDefName));
+                    break;
+                case CALLBACK:
+                    validateAction(((CallbackState) state).getAction(), functionDefName);
+                    break;
+                // Sonar compliance...
+                default:
+                    break;
```

Since you add the default (sonar thing), I think you can throw an unsupportedoperationException here for completeness

---

Makes sense.

---

@fjtirado, so the custom work item handler won't call the add-on if there's no action definition in the state? Otherwise, we will throw exceptions for no reason.